### PR TITLE
Adds staff of polymorph, removes staff of change from available wizard artifacts

### PR DESCRIPTION
**Pull request by MadmanMartian**
Considering how easy it is to turbo'gunk' a station using the staff of change, I've opted to make an alternate that's 'just as fun' but not as round destroying as setting the staff of change to 'xeno'.

 - Wand of Polymorph
     - Two settings: Minor and Major. Minor turns the afflicted into a harmless simple animal for a short time, Major turns the afflicted into a somewhat more dangerous animal for a longer period of time
      - Trying to shoot yourself with the wand of polymorph turns you into a...

 - Pollywog
     - A strange mass of alchemical inconsistency, that transmogrifies every so often into something else.

:cl:
 * rscadd Adds the Staff of Polymorph, available via wizard spellbooks
 * rscdel: Removes the staff of change from the wizard spellbooks

### CODE REVIEW

**Comment by Shadowmech88** [comment]
Do pollywogs retain their wizard spells?

**Comment by MadmanMartian** [comment]
transmogrify doesn't call transfer_mind, so I don't believe it will.

**Comment by KingofHONK** [comment]
would upboat, but I enjoy xenos and would hate to see them go away for the unforeseeable future

**Comment by DrSnips** [comment]
I upvoted because I dislike xenos. Having something most people consider a round ending mob as an option was a huge mistake.

**Comment by SonixApache** [comment]
👍 for fun new staff
👎 for remove turbogunker in same PR

**Comment by DrSnips** [comment]
I'd like to try it out as is right now as well as the pollywog. It seems interesting.
@MadmanMartian The one mob I have an issue with in the list of possible mobs is mice because they basicly have an auto-death feature where they have to constantly eat or they die. This was added a long time ago due to something with spectators becoming mice.

**Comment by MadmanMartian** [comment]
Death isn't much of a problem when you're polymorphed

If the polymorph wand is at minor or major, once you die you just come back as yourself.

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -2060,7 +2060,7 @@ mob/proc/on_foot()
 	spawn(duration + 1)
 		regenerate_icons()
 
-/mob/proc/transmogrify(var/target_type, var/offer_revert_spell = FALSE)	//transforms the mob into a new member of the given mob type, while preserving the mob's body
+/mob/proc/transmogrify(var/target_type, var/offer_revert_spell = FALSE, var/kill_on_death = TRUE, var/allow_revert = TRUE)	//transforms the mob into a new member of the given mob type, while preserving the mob's body
```
Comment (Shadowmech88) [inline code/modules/mob/mob.dm]: Why add an argument for making a transmogrification permanent? All of the utility is in being able to undo the transformations via code if necessary.
All of the code that is run when `allow_revert` is `TRUE` is essentially
```
var/mob/M = new type(loc)
M.key = key
qdel(src)
```
In which case you could just use those lines instead, or make a new proc specifically to permanently move a player into a new instance of a given mob if you'd really rather use 1 line than 3.

File: code/modules/projectiles/projectile/change.dm [hunk 2]
```diff
@@ -54,3 +54,45 @@
...
+	flag = "energy"
+	fire_sound = 'sound/weapons/radgun.ogg'
+	var/status = MAJOR
+
+/obj/item/projectile/polymorph/on_hit(var/atom/hit)
+	if(hit == firer)
+		firer.transmogrify(/mob/living/simple_animal/pollywog, allow_revert = FALSE)
+		return
+	var/stat = status
+	spawn(1)//fixes bugs caused by the target ceasing to exist before the projectile has died.
```
Comment (Shadowmech88) [inline code/modules/projectiles/projectile/change.dm]: Could this be fixed by manually calling whatever proc kills the projectile right here, before running `polymorph()`?

**Comment by Shadowmech88** [comment]
Is the pollywog supposed to be a legitimate use of the wand, or just an interesting quirk for people who are attempting to shoot themselves in the head anyway?
If the former, I think the combination of unpredictability, a lack of spells, and permanent death makes things unnecessarily hard on the wizard.

Also, I'm not sure how useful the wand will be even on other people. They don't die when their new mobs do, which is fine, but it means the wand can't be used to weaken opponents because killing them won't do anything. Additionally, such a short time until untransmogrifying means that it will be difficult to use this wand for chaos, as any department you upset will be back to normal in less than a minute.
I can't really see any applications for this as-is besides giving allies a coat of `simple_animal` armor for 10-30 seconds, and only because doing so couldn't possibly hurt. But I don't know if anyone will pay 40 spell points for that benefit.

**Comment by boy2mantwicethefam** [comment]
I like the idea but two things:
- Nothing prevents the transformed person to attack the wizard using their shiny new animal body that is immune to stuns, so almost everyone will go with minor transformations
- If Staff of Change is such a big deal because it can spawn xenos and xenos are round-enders, why not just remove xenos from the options?

I'd say add wand of polymorph as a separate thing and maybe increase the price of the staff of change without removing the xenos

**Comment by DrSnips** [comment]
We already tried raising the cost of SoC due to xenos and other mobs it can make. That didn't solve the xeno issue in the slightest.

**Comment by Shadowmech88** [comment]
Why not just restrict the types of xenos it can make to the non-reproducing ones?

**Comment by DrSnips** [comment]
Because those are still xenos.

**Comment by ShiftyRail** [comment]
Xenos are fine as a round ender
The staff of gunk itself isn't too bad, it's just that coupled with a stun aimbot (magic missile) or blink it's unfun to play against
Make it require more spell points or have an higher cooldown, or better, a limited amount of Benos spawned at a time

**Comment by DrSnips** [comment]
That still makes/introduces xenos and doesn't solve the issue.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
This is a massive overnerf, in my opinion

**Comment by Shadowmech88** [comment]
Are xenos really a round-ending threat if their number is restricted to only what the wizard creates manually?

**Comment by DrSnips** [comment]
Yes and no. They can make more xenos if they are allowed to become drones/queens. They can acid pretty much qnything and by simply aciding the comms consoles and boards, research console, and cargo for good measure they effectively make the round xeno-stended. I've seen stealth xenos before where they just vent crawled nearly forever, stocked up chems, and removed things slowly. Overall they introduce unhealthy gameplay with qdel acid, not truely stunable, ventcrawl, hyperspeed, almost for sure disarms that can even stun silicons, near invisibility, the ability to touch most electronics and disable them, needs no atmos, and have funky resists.

**Comment by SonixApache** [comment]
Does this work in turrets?

**Comment by Blithering** [comment]
How about removing the option to choose conversion so it becomes a lot more RNG based? You could then add in limited use wands with specific transformations on them for gimmick wizards to use.  I do like pollywogs though.

**Comment by Hoshi-chan** [comment]
Add it, but increase the cost of the staff of change instead of removing it outright.

The idea of just making the staff of change always random is good too.

**Comment by jknpj** [comment]
I'll close it for now since it has the derelict tag and all.

**Comment by Blithering** [comment]
RIP, Pollywogs as a separate PR soon?

**Pull request closed without merging**